src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/semantic/validator/definition/RequiredPropertiesValidator.java

```diff
@@ -1,111 +0,0 @@
-            return;
-        }
-
-        if (!properties.contains(discriminator)) {
-            validationErrors.add(new DefinitionSemanticError(holder.getCurrentPath(),
-                    "discriminator \"" + discriminator + "\" is not a property defined at this schema"));
-        }
-
-        if (!requiredProperties.contains(discriminator)) {
-            validationErrors.add(new DefinitionSemanticError(holder.getCurrentPath(),
-                    "discriminator property \"" + discriminator + "\" is not marked as required"));
-        }
-    }
-
-    private void validateProperties(Collection<String> objectProperties, List<String> required) {
-        Set<String> duplicates = Util.findDuplicates(required);
-        if (!duplicates.isEmpty()) {
-            validationErrors.add(new DefinitionSemanticError(holder.getCurrentPath(),
-                    "required property is defined multiple times: " + duplicates));
-        }
-
-        if (objectProperties.containsAll(required)) {
-            return;
-        }
-
-        List<String> requiredButNotDefinedProperties = new ArrayList<>(required);
-        requiredButNotDefinedProperties.removeAll(objectProperties);
-        validationErrors.add(new DefinitionSemanticError(holder.getCurrentPath(),
-                "required properties are not defined as object properties: " + requiredButNotDefinedProperties));
-    }
-
-    @Override
-    public void validate(ArrayModelWrapper arrayModelWrapper) {
-        if (arrayModelWrapper.getModel().getItems() == null) {
-            validationErrors
-                    .add(new DefinitionSemanticError(holder.getCurrentPath(), "'items' must be defined for an array"));
-            return;
-        }
-        arrayModelWrapper.getItems().accept(this);
-    }
-
-    @Override
-    public void validate(ComposedModelWrapper composedModelWrapper) {
-        for (Model element : composedModelWrapper.getAllOf()) {
-            String modelName = element instanceof RefModel ? ((RefModel) element).getSimpleRef() : "child";
-            VisitableModelFactory.createVisitableModel(modelName, element).accept(this);
-        }
-    }
-
-    @Override
-    public void validate(ObjectPropertyWrapper objectProperty) {
-        validateProperties(objectProperty.getProperties().keySet(), objectProperty.getRequiredProperties());
-
-        for (Map.Entry<String, Property> property : objectProperty.getProperties().entrySet()) {
-            VisitablePropertyFactory.createVisitableProperty(property.getKey(), property.getValue()).accept(this);
-        }
-    }
-
-    @Override
-    public void validate(ArrayPropertyWrapper arrayProperty) {
-        if (arrayProperty.getProperty().getItems() == null) {
-            validationErrors
-                    .add(new DefinitionSemanticError(holder.getCurrentPath(), "'items' must be defined for an array"));
-            return;
-        }
-
-        arrayProperty.getItems().accept(this);
-    }
-}
```

Comment:
Does removal of the file will not cause backward compatibility issues?

Reply:
Unless someone have extended the RequiredPropertiesValidator in their code, I don't see other compatibility issues